### PR TITLE
PaginationService doesn't support \n

### PR DESCRIPTION
**Pull request by rexbut**
The pagination service doesn't support the character "\n". With the character " \n", the number of line is different to the real number of line.

- **Before :**

Error_1
Error_2

- **After :**

Fix_1
Fix_2

### CODE REVIEW

File: src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java [hunk 1]
```diff
@@ -163,8 +163,13 @@ int getWidth(Text text) {
             Integer cp;
             while(i_it.hasNext()){
                 cp = i_it.next();
-                int width = getWidth(cp, bold);
-                total += width;
+                //The newline character is 10
+                if (cp == 10) {
```
Comment (simon816) [inline src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java]: Would be more clear to use `'\n'` here instead of `10`

**Review by windy1 — DISMISSED** [review]
Just tested this. The issue I found looks to be that you don't iterate over the text children deeply, meaning it will only pick up newlines on the top children level.

**Comment by stephan-gh** [comment]
@windy1 That is weird because the code uses `Text.withChildren()` which provides an recursive iterator over all children of a Text instance. It should process all children correctly.

**Comment by windy1** [comment]
I could be wrong then, I wasn't aware of that and that was really more of a (not so) educated guess, however I tested with nested text with newlines in them and it didn't work :/

**Comment by rexbut** [comment]
@windy1 Didn't worked when there was /n several times.

File: src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java [hunk 2]
```diff
@@ -161,10 +161,22 @@ int getWidth(Text text) {
                     .getBold();
 
             Integer cp;
+            boolean new_line = false;
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java]: This should be called `newLine`, not `new_line`.

**Review by kashike — CHANGES_REQUESTED** [review]

**Review by kashike — APPROVED** [review]

**Comment by randombyte-developer** [comment]
This also seems to fix an issue with headers(probably footers too): When having a multiline header the pages get to long. The fixed `getWidth()` should fix that.

**Pull request merged**